packages/smooth_app/lib/pages/prices/price_category_input_page.dart

```diff
@@ -0,0 +1,285 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:smooth_app/generic_lib/design_constants.dart';
+import 'package:smooth_app/generic_lib/dialogs/smooth_alert_dialog.dart';
+import 'package:smooth_app/pages/input/smooth_autocomplete_text_field.dart';
+import 'package:smooth_app/pages/prices/price_meta_product.dart';
+import 'package:smooth_app/pages/product/may_exit_page_helper.dart';
+import 'package:smooth_app/pages/product/simple_input/simple_input_page_helpers.dart';
+import 'package:smooth_app/query/product_query.dart';
+import 'package:smooth_app/widgets/smooth_scaffold.dart';
+import 'package:smooth_app/widgets/will_pop_scope.dart';
+
+const EdgeInsets _fieldPadding = EdgeInsets.symmetric(
+  horizontal: LARGE_SPACE,
+  vertical: MEDIUM_SPACE,
+);
+const BorderRadius _borderRadius = CIRCULAR_BORDER_RADIUS;
+
+/// Page that lets the user type in and select a single category for prices.
+class PriceCategoryInputPage extends StatefulWidget {
+  const PriceCategoryInputPage();
+
+  @override
+  State<PriceCategoryInputPage> createState() => _PriceCategoryInputPageState();
+}
+
+class _PriceCategoryInputPageState extends State<PriceCategoryInputPage> {
+  late final TextEditingController _categoryController;
+  late final TextEditingController _originController;
+
+  String? _categoryName;
+  final List<String> _originNames = <String>[];
+  bool _hasChanged = false;
+
+  late final VoidCallback _changes;
+
+  @override
+  void initState() {
+    super.initState();
+    _changes = () {
+      _hasChanged = true;
+    };
+    _categoryController = TextEditingController();
+    _categoryController.addListener(_changes);
+    _originController = TextEditingController();
+    _originController.addListener(_changes);
+  }
+
+  @override
+  void dispose() {
+    _categoryController.removeListener(_changes);
+    _categoryController.dispose();
+    _originController.removeListener(_changes);
+    _originController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context);
+    return WillPopScope2(
```

Comment:
Please implement the same as in #6590, otherwise the swipe gesture is always blocked

Reply:
@g123k Shouldn't we extend that "swipe gesture fix" to all files that use `WillPopScope2`?

Reply:
All other screens are OK.
This is only one missing the Controller thing

Reply:
@g123k I'm puzzled. I checked this morning and not all WillPopScope2 files used that new controller.
Would that mean that some WillPopScope2 need the controller, and some don't?
If so the typical cases should be documented in the new controller class. Maybe they are - I cannot check right now.